unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs

```diff
@@ -631,7 +632,16 @@ public static Boolean PassModifierKeys(KeyCode[] modifierKeys)
             }
             else
             {
-                if (modifierKeys.Contains(keyCode) != pressed)
+                // Very confusing function, but this should avoid allocations and LINQ
+                var contains = false;
+                foreach (KeyCode kc in modifierKeys)
+                {
+                    if (kc == keyCode)
+                    {
+                        contains = true;
+                    }
+                }
+                if (contains != pressed)
```

Comment:
Another nitpick for readability

```suggestion
                var anyModifierKeysActive = false;
                foreach (KeyCode kc in modifierKeys)
                {
                    if (kc == keyCode)
                    {
                        anyModifierKeysActive = true;
                        break;
                    }
                }
                if (anyModifierKeysActive != pressed)
```